### PR TITLE
Update usage examples

### PR DESCRIPTION
**Pull request by cole**
Try to cover a few more basic use cases; also, test our example code.

Connects to https://github.com/archivematica/Issues/issues/609, though it's been verified already! I should have been clearer there were a few more cleanups remaining.

### CODE REVIEW

File: docs/reading-mets.rst [hunk 1]
```diff
@@ -18,27 +18,82 @@ metsrw supports reading METS files from disk, from strings, or from lxml_
...
-    mets.get_file(file_uuid=file_uuid)
+    >>> mets.get_file(file_uuid=file_uuid)
+    FSEntry(type='Item', path='test/hello.pdf', use='original', ...)
+
+    >>> mets.all_files()
+    {FSEntry(type='Item', path='test/hello.pdf', use='original', ...)}
+
+    # Currently, filtering files can only be done via iteration
+    >>> [entry for entry in mets.all_files() if entry.use == "original"]
+    [FSEntry(type='Item', path='test/hello.pdf', use='original', ...)]
```
Comment (cole) [inline docs/reading-mets.rst]: @replaceafill should cover your use case here:
```
Iterate file entries in the <fileSec> according to their @USE:
mets = metsrw.METSDocument.fromfile(path)
original_files = [entry for entry in mets.all_files() if entry.use == "original"]```

File: docs/reading-mets.rst [hunk 2]
```diff
@@ -18,27 +18,82 @@ metsrw supports reading METS files from disk, from strings, or from lxml_
...
+.. doctest::
+
+    # Currently, filtering PREMIS objects can only be done via iteration
+    >>> ingestion_events = []
+    >>> mets = metsrw.METSDocument.fromfile('../fixtures/complete_mets.xml')
+    >>> for fsentry in mets.all_files():
+    ...     for event in fsentry.get_premis_events():
+    ...          if event.type == "ingestion":
+    ...              ingestion_events.append(event)
+    >>> ingestion_events[0]
```
Comment (cole) [inline docs/reading-mets.rst]: @replaceafill should cover this use case:
```
Filter PREMIS events of a file entry by type:
# entry is a FSEntry object
reingestion_events = [
    event for event in entry.get_premis_events() if event.type == 'reingestion'
]
```

File: docs/reading-mets.rst [hunk 3]
```diff
@@ -18,27 +18,82 @@ metsrw supports reading METS files from disk, from strings, or from lxml_
...
+`amdSec` and `dmdSec` data is accessible via the
+:attr:`~metsrw.FSEntry.amdsecs` and :attr:`~metsrw.FSEntry.dmdsecs`
+properties.
+
+.. doctest::
+
+    >>> mets = metsrw.METSDocument.fromfile('../fixtures/complete_mets.xml')
+    >>> fsentry = mets.get_file(file_uuid="ab5c67fc-8f80-4e46-9f20-8d5ae29c43f2")
+    >>> amdsec1 = fsentry.amdsecs[0]
+    >>> [section for section in amdsec1.subsections if section.subsection == 'techMD']
```
Comment (cole) [inline docs/reading-mets.rst]: @replaceafill  should cover this use case:
```
Use the METS tree to get the <amdSec> from a file entry:
# entry is a FSEntry object
amdSec = mets.tree.find(
    'mets:amdSec[@ID="{}"]'.format(entry.admids[0]),
    namespaces={...},
)
```

I think it's probably better to just get the PREMIS data directly, so I've added the note below.

File: docs/reading-mets.rst (hunk 1 above)
Comment (replaceafill) [reply docs/reading-mets.rst]: :+1:

File: docs/reading-mets.rst (hunk 2 above)
Comment (replaceafill) [reply docs/reading-mets.rst]: :+1:

File: docs/reading-mets.rst (hunk 3 above)
Comment (replaceafill) [reply docs/reading-mets.rst]: Ah! Good suggestion. Thank you @cole!

**Pull request merged**